File: C_Sharp/InterfacesLab/InterfacesLab/SonicScholarLawnCareProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace InterfacesLab
{
    class SonicScholarLawnCareProvider : ILawnCareProvider
    {
        int _totalJobCount;
        public SonicScholarLawnCareProvider()
        {
            _totalJobCount = 0;
        }

        public void BookMowing()
        {
            _totalJobCount++;
        }

        public void BookRaking()
        {
            _totalJobCount++;
        }
        public void BookWeedRemoval()
        {
            _totalJobCount++;
        }

        public bool IsAvailable
        {
            get
            {
                return _totalJobCount < 3;
            }
        }

        public int Rating
        {
            get
            {
                return 5;
            }
        }

        public int TotalJobCount
        {
            get
            {
                return _totalJobCount;
            }
        }
    }
}

File: Java/InterfaceLab/src/com/sonicscholar/SonicScholarLawnCareProvider.java
package com.sonicscholar;

public class SonicScholarLawnCareProvider implements ILawnCareProvider {
    double _totalJobCount;

    public SonicScholarLawnCareProvider(){
        _totalJobCount = 0;
    }

    public void bookMowing(){
        _totalJobCount++;
    }

    public void bookRaking(){
        _totalJobCount++;
    }

    public void bookWeedRemoval(){
        _totalJobCount++;
    }

    public boolean isAvailable(){
        return _totalJobCount < 3;
    }

    public int getRating(){ return 5;}

    public double get_totalJobCount() {
        return _totalJobCount;
    }
}
